### PR TITLE
Fixes Custom Header being ignored

### LINKED ISSUE #272: odata-cli custom header not used when fetching metadata
### Describe the bug

I'm trying to create OData classes using the new CLI (great idea, I love it). My service is protected and thus needs a custom header. This includes the metadata call, not only the actual call to execute the request!

#### Version of the Project affected
Microsoft.OData.Cli v0.1.0

#### To Reproduce

Cal the CLI generate command for a service protected by a custom header.

```odata-cli.exe generate -m "https://my-server/my-service/$metadata" -o output --multiple-files -h Authorization:"Basic ......." -ns "service.ns" -fn "Odata"```

#### Expected behavior

No `The remote server returned an error: (401) Unauthorized.`

#### Actual behavior

`The remote server returned an error: (401) Unauthorized.`

#### Additional context

I've had a look, it seems like the custom headers aren't used here. The webRequest uses proxy and everything, but here the custom header isn't used:
https://github.com/OData/ODataConnectedService/blob/master/src/Microsoft.OData.CodeGen/Common/MetadataReader.cs#L51-L69

Is there a reason why the custom headers aren't used there?

### PR DESCRIPTION
**Pull request by ddprince-yaksa**
The OData Cli is ignoring the custom headers at the first step when obtaining the metadata information. This pull request fixes this issue. Also, I've taken care of fixing a misleading command line argument comment. 

This fixes #272

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OData/ODataConnectedService?pullRequest=273) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OData/ODataConnectedService?pullRequest=273) before we can accept your contribution.<br/><br/>:x: ddprince-yaksa [sign now](https://cla.dotnetfoundation.org/OData/ODataConnectedService?pullRequest=273)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OData/ODataConnectedService?pullRequest=273) it.</sub>

**Comment by ddprince-yaksa** [comment]
Hello, 

I signed the second CLA notice, but it's still showing that I did not: 
image

Thank you,
David

File: src/Microsoft.OData.CodeGen/Common/MetadataReader.cs [hunk 1]
```diff
@@ -106,5 +108,24 @@ public static Version GetMetadataVersion(ServiceConfiguration serviceConfigurati
...
+        {
+            if (string.IsNullOrWhiteSpace(serviceConfiguration.CustomHttpHeaders)) 
+                return;
+            
+            var customHeaders = serviceConfiguration.CustomHttpHeaders.Split(',');
+            foreach (var customHeader in customHeaders)
+            {
+                var headerKeyValuePair = customHeader.Split(':');
+
+                if (headerKeyValuePair.Length > 1)
```
Comment (KenitoInc) [inline src/Microsoft.OData.CodeGen/Common/MetadataReader.cs:122]: For code maintainability, wrap if and else in curly braces

Comment (ddprince-yaksa) [reply src/Microsoft.OData.CodeGen/Common/MetadataReader.cs:122]: Hello @KenitoInc , I just did the modifications. Please let me know if there is anything else! 
Thank you!

**Comment by ddprince-yaksa** [comment]
Hello, do you have an update regarding this pull request? 

Thanks!

**Comment by ElizabethOkerio** [comment]
There is a PR out with the fix: https://github.com/OData/ODataConnectedService/pull/302. We hope to have it merged soon

**Comment by Nthemba** [comment]
> There is a PR out with the fix: #302. We hope to have it merged soon

PR #311, that fixes this issue is merged.

**Pull request closed without merging**